labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -45,7 +57,10 @@ object RecursionPatternMatchingExercise {
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
   def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+    in match {
+      case (a :: tail) => (a :: tail.filter((t:T) => t == a)) :: groupEquals (tail.filter((t:T) => t != a))
+      case _ => List()
+    }
   }
```

Comment:
I like your use of conditions (t !=a) for filtering in this function. I had the same condition but my case was structured in an entirely different way, with the condition closer to the beginning of the statement. It's cool that we can do something so similar but have a completely different implementation.